### PR TITLE
Move obcpbft events infrastructure into its own package

### PR DESCRIPTION
**Pull request by jyellick**
## Description

This changeset moves the obcpbft events framework into its own package and fixes all of the corresponding local references in the code to be package references.

It also renames the assorted event methods to not stutter under the new package naming scheme.

It further switches from using the generic `interface{}` for events to the typed alias `Event`.
## Motivation and Context

This is part of the ongoing effort to de-multithread the pbft consensus system.  This move was suggested by @corecode when the events framework was introduced.

In order to remove one of the final go routines from `consensus/obcpbft/custodian` this package must be modified to utilize the new events framework.  In order to do this in a way without cyclical references, the events framework must be moved into its own package.
## How Has This Been Tested?

This changeset contains absolutely no logical code changes, it is purely semantic, so existing CI should cover testing.
## Checklist
- [X] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [X] Either no new documentation is required by this change, OR I added new documentation
- [X] Either no new tests are required by this change, OR I added new tests
- [X] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Jason Yellick jyellick@us.ibm.com

### CODE REVIEW

**Comment by corecode** [comment]
go for it

File: docs/source/chaincode4ade.rst [hunk 1]
```diff
@@ -84,319 +61,479 @@ To keep things simple, let's use the following command:
...
-    )
+  // SmartContract provides functions for managing an Asset
+  type SmartContract struct {
+  contractapi.Contract
+  }
 
-    // SimpleAsset implements a simple chaincode to manage an asset
-    type SimpleAsset struct {
-    }
+Next, let's add a struct ``Asset`` as a receiver for Chaincode functions.
```
Comment (denyeart) [inline docs/source/chaincode4ade.rst:97]:
```suggestion
Next, let's add a struct ``Asset`` to represent simple assets on the ledger. Note the JSON annotations, which will be used to marshal the asset to JSON which is stored on the ledger.
```

File: docs/source/chaincode4ade.rst [hunk 2]
```diff
@@ -84,319 +61,479 @@ To keep things simple, let's use the following command:
...
+    "fmt"
+    "log"
+    "github.com/hyperledger/fabric-contract-api-go/contractapi"
+  )
 
-    	"github.com/hyperledger/fabric-chaincode-go/shim"
-    	"github.com/hyperledger/fabric-protos-go/peer"
-    )
+  // SmartContract provides functions for managing an Asset
+  type SmartContract struct {
```
Comment (denyeart) [inline docs/source/chaincode4ade.rst:93]: indent

File: docs/source/chaincode4ade.rst [hunk 3]
```diff
@@ -84,319 +61,479 @@ To keep things simple, let's use the following command:
...
+    Size           int    `json:"size"`
+    Owner          string `json:"owner"`
+    AppraisedValue int    `json:"appraisedValue"`
+  }
 
 Initializing the Chaincode
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Next, we'll implement the ``Init`` function.
+Next, we'll implement the ``Init`` function to populate the ledger with some initial data.
```
Comment (denyeart) [inline docs/source/chaincode4ade.rst:113]:
```suggestion
Next, we'll implement the ``InitLedger`` function to populate the ledger with some initial data.
```

File: docs/source/chaincode4ade.rst [hunk 4]
```diff
@@ -84,319 +61,479 @@ To keep things simple, let's use the following command:
...
+      }
+    }
+
+    return nil
   }
 
 .. note:: Note that chaincode upgrade also calls this function. When writing a
           chaincode that will upgrade an existing one, make sure to modify the ``Init``
           function appropriately. In particular, provide an empty "Init" method if there's
           no "migration" or nothing to be initialized as part of the upgrade.
```
Comment (denyeart) [inline docs/source/chaincode4ade.rst:146]: Remove the info about Init being called for upgrade.

**Pull request merged**